watertap/unit_models/MD/membrane_distillation_base.py

```diff
@@ -306,12 +500,31 @@ def _add_mass_flux(self):
             doc="Solvent mass flux",
         )
         def eq_flux_mass(b, t, x):
-            return b.flux_mass[t, x] == b.permeability_coef[
-                t
-            ] / b.membrane_thickness * (
-                b.hot_ch.properties_interface[t, x].pressure_sat
-                - b.cold_ch.properties_interface[t, x].pressure_sat
-            )
+
+            if self.config.MD_configuration_Type == MDconfigurationType.DCMD:
+                return b.flux_mass[t, x] == b.permeability_coef[
+                    t
+                ] / b.membrane_thickness * (
+                    b.hot_ch.properties_interface[t, x].pressure_sat
+                    - b.cold_ch.properties_interface[t, x].pressure_sat
+                )
+            elif self.config.MD_configuration_Type in [
+                MDconfigurationType.PGMD_CGMD,
+                MDconfigurationType.AGMD,
+            ]:
+                return b.flux_mass[t, x] == b.permeability_coef[
+                    t
+                ] / b.membrane_thickness * (
+                    b.hot_ch.properties_interface[t, x].pressure_sat
+                    - b.gap_ch.properties_interface[t, x].pressure_sat
+                )
+            elif self.config.MD_configuration_Type == MDconfigurationType.VMD:
+                return b.flux_mass[t, x] == b.permeability_coef[
+                    t
+                ] / b.membrane_thickness * (
+                    b.hot_ch.properties_interface[t, x].pressure_sat
+                    - b.cold_ch.properties[t, x].pressure
+                )
```

Comment:
Is this constraint different from the one used above for DCMD?

Reply:
Yes, in VMD, there is only vapor phase which is fixed at vacuum pressure, and there is a single main control volume state block. In DCMD, there are separate state blocks for the interface and the vapor phase, and the main control volume state block, where the vapor pressure equals the saturation pressure at the interface.

Reply:
Ah, just noticed the discrepancy between cold channel `properties` vs. `properties_interface` between VMD and DCMD, respectively. Thanks!